ui/pager.go

```diff
@@ -199,6 +202,92 @@ func (m pagerModel) Update(msg tea.Msg) (pagerModel, tea.Cmd) {
 			}
 		default:
 			switch msg.String() {
+			case "e":
+				// First, determine if we can find an editor
+				editor_path, err := getEditor()
+
+				if err != nil {
+					cmd := m.showStatusMessage("No editor found!")
+					cmds = append(cmds, cmd)
+					break
+				}
+
+				// Determine if the current file is stashed or not
+				isStashed := m.currentDocument.markdownType == stashedMarkdown ||
+					m.currentDocument.markdownType == convertedMarkdown
+				
+				// Copy the contents of the document to a temporary file
+				tempFile, err := ioutil.TempFile("", "glow-edit")
+				if err != nil {
+					cmd := m.showStatusMessage("Error creating temporary file!")
+					cmds = append(cmds, cmd)
+				} else {
+					// Ensure the temporary file gets deleted
+					defer os.Remove(tempFile.Name())
+				}
+				tempFile.WriteString(m.currentDocument.Body)
+				tempFile.Close()
+
+				// Use our editor to edit the temporary file
+				cmd := exec.Command(editor_path, tempFile.Name())
+				cmd.Stdin = os.Stdin
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				err = cmd.Start()
+				if err != nil {
+					log.Println(err)
+				}
+				err = cmd.Wait()
+				if err != nil {
+					log.Println(err)
+				}
+
+				// Get the contents of the edited file
+				b, err := ioutil.ReadFile(tempFile.Name())
+				if err != nil {
+					log.Println(err)
+					break
+				}
+				editString := string(b)
+				
+				// If no edits were made, do nothing
+				if editString == m.currentDocument.Body {
+					cmd := m.showStatusMessage("No changes made!")
+					cmds = append(cmds, cmd)
+				} else {
+					// Add our edits to the current view
+					m.currentDocument.Body = editString
+					
+					// Re-render current doc
+					cmd := renderWithGlamour(m, m.currentDocument.Body)
+					cmds = append(cmds, cmd)
+
+					// Make our edits permanent remotely/locally
+					if isStashed {
+						// This method is preferable, pending a PR for charm
+						//err := m.general.cc.SetMarkdownBody(m.currentDocument.ID, editString)
+
+						// Create a new stash
+						_, err := m.general.cc.StashMarkdown(m.currentDocument.Note, editString)
+
+						if err != nil {
+							cmd := m.showStatusMessage("Couldn't update stash!")
+							cmds = append(cmds, cmd)
+						} else {
+							// Delete the current stash
+							m.general.cc.DeleteMarkdown(m.currentDocument.ID)
+							cmd := m.showStatusMessage("Updated stash!")
+							cmds = append(cmds, cmd)
+						}
+					} else {
+						err := ioutil.WriteFile(m.currentDocument.localPath, []byte(editString), 0777)
```

Comment:
Ideally we look at the file's original permissions and use the same ones again when we write here.

Reply:
Fair enough. I was under the impression that the permissions were basically ignored if the file already exists, and that parameter only does anything on creating a new file. I'll definitely add that check.

Reply:
@shitchell That's actually correct:

WriteFile writes data to a file named by filename. If the file does not exist, WriteFile creates it with permissions perm (before umask); otherwise WriteFile truncates it before writing, without changing permissions.

I still think we should probably specify 0600 as a reasonable default here.